lib/board.rb

```diff
@@ -0,0 +1,60 @@
+class Board
+  attr_accessor :cells
+
+  def initialize
+    @cells = []
+    reset!
+  end
+
+  def reset!
+    @cells.clear
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end  
+
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+  end
+
+  def position(number)
+    cells[number.to_i-1]
+  end
+
+  def full?
+    spot_filled = []
+    cells.each do |cell| 
+      if cell.include?("X") || cell.include?("O")
+        spot_filled << true
+      else 
+        spot_filled << false
+      end
+    end
+    ( spot_filled.include?(false) ? false : true ) 
+  end
```

Comment:
Use the most expressive and efficient iterator for a problem. `Board#full?` is asking if all elements are not empty strings. You can use logic that is symmetrical to those semantics with the [`#all?`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-all-3F) iterator.

```
def full?
  cells.all? { |cell| cell == "X" || cell == "O" }`
end
```

This is also more efficient in that the loop will stop after the iteration for which a cell is not empty.